### ISSUE TITLE
infinite loop?

### ISSUE THREAD

**Issue by jdbancal** [replab/replab#68]
The following commands seems to trigger an infinite loop in replab.bsgs.chain.wordsComplete
```
group = replab.SignedPermutations(12).subgroup({[1 3 2 5 6 7 4 9 8 12 10 11]})
group.naturalRep.decomposition
```

**Comment by denisrosset** [comment]
Indeed, the problem happens in the factorization method, and can be triggered by calling
`group.factorization(group.sample)`
a few times.

There is a bug in the implementation of the "short words" algorithm.

**Comment by denisrosset** [comment]
It's not in the nicest code paths of RepLAB, and the bug seems tricky to find. The BSGS implementation needs a clean up and documentation.
